### PR TITLE
Implement Txn Count endpoint

### PR DESCRIPTION
**Pull request by bholagabbar**

### CODE REVIEW

File: api/utils/chainUtils.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    let res: number[] = []
+    for (let key of Object.keys(getChains())) {
+        res.push(parseInt(key))
+    }
+    return res
+}
+
+export function getChainNameFromId(chainId: number|string) {
+    let chainIdStr: string = chainId.toString()
+    // @ts-ignore
```
Comment (bholagabbar) [inline api/utils/chainUtils.ts]: This keeps throwing me an error if I remove the ts-ignore. Is there a cleaner way ?

File: api/v1/routes/volume.ts [hunk 2]
```diff
@@ -9,13 +12,72 @@ router.get('/total/tx_count/:direction',
...
+                {
+                    $sort: {
+                        _id: 1
+                    }
+                }
+            ]).toArray()
+
+            // Maps date to count
+            let chainName = getChainNameFromId(chainId)
+            // @ts-ignore
```
Comment (bholagabbar) [inline api/v1/routes/volume.ts]: Multiple ts-ignores, nicer way or am I doing something wrong ?

File: index.ts [hunk 3]
```diff
@@ -9,26 +8,29 @@ import {MongoConnection} from "./api/db/MongoConnection";
 import {RedisConnection} from "./api/db/RedisConnection";
 import volume from "./api/v1/routes/volume";
 
-// Connect to DBs
-async () => {
+async function setupDBs() {
```
Comment (bholagabbar) [inline index.ts]: Can't find a cleaner way to bootstrap the API only after the DBs have been created. async/await in root isn't allowed in typescript here

**Comment by trajan0x** [comment]
@metagabbar you can get rid of the `ts-ignore` [here](https://github.com/synapsecns/synapse-analytics-api/blob/27bf7e15b535001ee4ed8903d263da0ad7fbccd4/api/db/RedisConnection.ts#L30) by changing the function signature to 

`static async setForQuery(queryName: String, args: Object, mongoResults: Object, expireIn: number)` (so `Number`->`number`)

That's because `Number` is a strict superset of number (defined in the function signature here:

image

**Comment by trajan0x** [comment]
Going forward let's follow the feature branch naming convention: `feat/volume`

File: api/utils/chainUtils.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+export function getChainIdNums() {
+    let res: number[] = []
+    for (let key of Object.keys(getChains())) {
+        res.push(parseInt(key))
+    }
+    return res
+}
+
+export function getChainIdStr() {
```
Comment (trajan0x) [inline api/utils/chainUtils.ts]: Is there a reason this is here if it's not being used?

File: api/utils/chainUtils.ts (hunk 1 above)
Comment (trajan0x) [reply api/utils/chainUtils.ts]: The problem here is you're using an enum to query a dictionary, so if we get:

```typescript
undefined === getChainNameFromId("999")
undefined != string
```

This method doesn't actually return a string

File: api/v1/routes/volume.ts [hunk 5]
```diff
@@ -1,21 +1,89 @@
...
+        }
 
-    return res.json({"data":{"ethereum" : []}})
+        // Determine direction and chainIds to find results for
+        let direction = req.params.direction
+        let chainIdsToFind: number[] = req.query.chain ?
+            [getChainIdNumFromName(req.query.chain)] : getChainIdNums()
 
-});
+        let resObj = {'data': {}}
```
Comment (trajan0x) [inline api/v1/routes/volume.ts]: Can we define the return type here, maybe in the model?

File: api/v1/routes/volume.ts (hunk 2 above)
Comment (trajan0x) [reply api/v1/routes/volume.ts]: see above, chainname could be undefined in the line above.

There aren't many circumstances where ts-ignore should be used

File: index.ts (hunk 3 above)
Comment (trajan0x) [reply index.ts]: I think bootstrapping services is fine here

**Review by trajan0x — CHANGES_REQUESTED** [review]

**Pull request merged**